ratis-client/src/main/java/org/apache/ratis/client/api/AsyncApi.java

```diff
@@ -45,14 +46,24 @@ default CompletableFuture<RaftClientReply> sendReadOnly(Message message) {
     return sendReadOnly(message, null);
   }
 
+  /** The same as sendReadOnly(message, server, true). */
+  default CompletableFuture<RaftClientReply> sendReadOnly(Message message, RaftPeerId server) {
+    return sendReadOnly(message, server, true);
+  }
+
   /**
    * Send the given readonly message asynchronously to the raft service.
+   * <p>
+   * The messages may be either ordered or unordered.
+   * For ordered messages, the reply futures are completed in the same order of the messages being sent.
+   * For unordered messages, the reply futures are completed in an arbitrary order.
    *
    * @param message The request message.
    * @param server The target server.  When server == null, send the message to the leader.
+   * @param ordered Are the messages ordered?
    * @return a future of the reply.
    */
-  CompletableFuture<RaftClientReply> sendReadOnly(Message message, RaftPeerId server);
+  CompletableFuture<RaftClientReply> sendReadOnly(Message message, RaftPeerId server, boolean ordered);
```

Comment:
Perhaps add one more method:

```java
  default CompletableFuture<RaftClientReply> sendReadOnly(Message message, boolean ordered) {
    return sendReadOnly(message, null, true);
  }
```

Reply:
Or two `sendReadOnlyUnordered` since the `sendReadOnly` actually dispatches totally different code path based on the bool flag.

Reply:
> Or two sendReadOnlyUnordered ...

This is a good idea.  Let me update it.